kube/terraform/main.tf

```diff
@@ -0,0 +1,309 @@
+resource "kubernetes_namespace" "m3db_namespace" {
+  metadata {
+    name = "m3db"
+  }
+}
+
+resource "kubernetes_service" "etcd_service" {
```

Comment:
Does order it appears in the file enforce ordering of how the objects are created? Just wondering because technically the headless service needs to be up before the statefulset

Reply:
No, ordering doesn't matter in the file.

Reply:
Is there a way to enforce it? Per the [docs](https://v1-11.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#statefulsetspec-v1-apps) (emphasis mine):

> serviceName is the name of the service that governs this StatefulSet. This **service must exist before the StatefulSet**, and is responsible for the network identity of the set. Pods get DNS/hostnames that follow the pattern: pod-specific-string.serviceName.default.svc.cluster.local where "pod-specific-string" is managed by the StatefulSet controller.